include/psen_scan_v2/scanner_reply_msg.h

```diff
@@ -94,6 +93,16 @@ class ScannerReplyMsg
   //! If the message is refused, the returned value is 0xEB.
   //! If the CRC is not correct, the device will not send any message.
   uint32_t res_code_{ 0 };
+
+public:
+  class CRCMismatch : public std::runtime_error
+  {
+  public:
+    CRCMismatch(const std::string& msg = "CRC did not match") : std::runtime_error(msg)
+    {
+    }
+    virtual ~CRCMismatch() = default;
+  };
 };
```

Comment:
Please, state all public stuff before the private stuff. This way the reader can easily overview the public interface of the class. Furthermore, ensure that first all public methods are stated and then all public members, etc.

I suggest that you state the possible exceptions at the beginning of the class. Please, separate the function declaration and the definition to ease readability.